openpype/lib/abstract_load_template.py

```diff
@@ -0,0 +1,305 @@
+import os
+import re
+import avalon
+from openpype.settings import get_project_settings
+from openpype.lib import Anatomy, get_linked_assets, get_loaders_by_name,\
+    collect_last_version_repres
+
+
+class AbstractTemplateLoader(object):
+    """
+    Abstraction of Template Loader.
+
+    Properties:
+        template_path : property to get current template path
+
+    Methods:
+        import_template : Abstract Method. Used to load template,
+            depending on current dcc
+        get_template_nodes : Abstract Method. Used to query nodes acting
+            as placeholders. Depending on current dcc
+    """
+
+    def __init__(self, placeholder_class):
+
+        self.import_template(self.template_path)
+
+        loaders_by_name = get_loaders_by_name()
+
+        # Skip if there is no loader
+        if not loaders_by_name:
+            self.log.warning("There are no registered loaders.")
+            return
+
+        current_asset = avalon.io.Session["AVALON_ASSET"]
+        self.populate_template(current_asset,
+                               loaders_by_name,
+                               placeholder_class)
+
+    @property
+    def template_path(self):
+        """
+        Property returning template path. Avoiding setter.
+        Getting template path from open pype settings based on current avalon
+        session and solving the path variables if needed.
+
+        Returns:
+            str: Solved template path
+
+        Raises:
+            ValueError: No profile found from settings for current avalon
+                session
+            KeyError: Could not solve path because a key does not exists
+                in avalon context
+            IOError: Solved path does not exists on current filesystem
+        """
+        project = avalon.io.Session["AVALON_PROJECT"]
+        anatomy = Anatomy(project)
+        project_settings = get_project_settings(project)
+        current_dcc = avalon.io.Session["AVALON_APP"]
+        current_task = avalon.io.Session["AVALON_TASK"]
+        build_info = project_settings[current_dcc]['templated_workfile_build']
+        profiles = build_info['profiles']
+
+        for profile in profiles:
+            if current_task in profile['task_types']:
```

Comment:
You're comparing `AVALON_TASK` which is the task name with a profile that is looking for task types. When I try this on a task named `anim_char1` that is of `animation` type, it will fail, because in the settings you only give option to filter by task type.

Reply:
Solved.
Added the possibilty to filter by task type and by task name.
Also if no task type or no task name is given in template settings,
then any task type or any task name may use the template
(wich means if no task type and no task name is given, you could define a default template for a host)